pkg/process/net/common_linux.go

```diff
@@ -34,14 +33,13 @@ const (
 	conntrackCachedURL   = "http://unix/" + string(sysconfig.NetworkTracerModule) + "/debug/conntrack/cached"
 	conntrackHostURL     = "http://unix/" + string(sysconfig.NetworkTracerModule) + "/debug/conntrack/host"
 	ebpfBTFLoaderURL     = "http://unix/debug/ebpf_btf_loader_info"
-	netType              = "unix"
 )
 
 // CheckPath is used in conjunction with calling the stats endpoint, since we are calling this
 // From the main agent and want to ensure the socket exists
 func CheckPath(path string) error {
 	if path == "" {
-		return fmt.Errorf("socket path is empty")
+		return errors.New("socket path is empty")
```

Comment:
I think it is better to declare this err as a global var

Reply:
Why? It is not a sentinel error.

Reply:
I think it reduces the allocations, I read it somewhere, but can't find the source now. maybe in modern go versions this is already taken care by the compiler though

Reply:
where does the memoization take place? is it something to be introduced in a follow-up PRs?

Reply:
Yes, sorry. It will come in a follow-up PR to include `CheckPath` in the memoization.